tests/unit/test.program.js

```diff
@@ -480,21 +507,154 @@ describe('program.main', () => {
         runOptions: {
           loadJSConfigFile: fakeLoadJSConfigFile,
         },
-      })
-      .then(() => {
-        const options = fakeCommands.lint.firstCall.args[0];
-        // This makes sure that the config object was applied
-        // to the lint command options.
-        assert.equal(
-          options.selfHosted, configObject.lint.selfHosted);
-      });
+      }
+    );
+
+    const options = fakeCommands.lint.firstCall.args[0];
+    // This makes sure that the config object was applied
+    // to the lint command options.
+    assert.equal(
+      options.selfHosted, configObject.lint.selfHosted);
+  });
+
+  it('discovers config files', async () => {
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const configObject = {
+      lint: {
+        selfHosted: true,
+      },
+    };
+    // Instead of loading/parsing a real file, just return an object.
+    const fakeLoadJSConfigFile = sinon.spy(() => {
+      return configObject;
+    });
+
+    await execProgram(
+      ['lint'],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          discoverConfigFiles: async () => ['fake/config.js'],
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      }
+    );
+
+    const options = fakeCommands.lint.firstCall.args[0];
+    // This makes sure that the config object was applied
+    // to the lint command options.
+    assert.equal(
+      options.selfHosted, configObject.lint.selfHosted);
+  });
+
+  it('lets you disable config discovery', async () => {
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+
+    const discoverConfigFiles = sinon.spy(() => Promise.resolve([]));
+    await execProgram(
+      ['--no-config-discovery', 'lint'],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          discoverConfigFiles,
+        },
+      }
+    );
+
+    sinon.assert.notCalled(discoverConfigFiles);
+  });
+
+  it('applies config files in order', async () => {
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+
+    const globalConfig = 'home/dir/.web-ext-config.js';
+    const projectConfig = 'project/dir/web-ext-config.js';
+    const customConfig = path.resolve('custom/web-ext-config.js');
+
+    const loadJSConfigFile = makeConfigLoader({
+      configObjects: {
+        [globalConfig]: {
+          noInput: true,
+        },
+        [projectConfig]: {
+          verbose: true,
+        },
+        [customConfig]: {
+          lint: {
+            selfHosted: true,
+          },
+        },
+      },
+    });
+
+    await execProgram(
+      ['lint', '--config', customConfig],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          discoverConfigFiles: async () => [
+            globalConfig, projectConfig,
+          ],
+          loadJSConfigFile,
+        },
+      }
+    );
+
+    const options = fakeCommands.lint.firstCall.args[0];
```

Comment:
This tests should test that the config file are loaded and applied in the expected order, how about wrapping `loadJSConfigFile` in a sinon.spy and assert that it has been called with the filenames in the expected order?

Reply:
Oh, yeah, thanks. This test wasn't checking the order at all. Fixed.